playwright-tests/testUtils.js

```diff
@@ -40,3 +42,28 @@ export const escapeHtml = (html) => {
     .replace(/"/g, "&quot;")
     .replace(/'/g, "&#039;");
 };
+
+export const useCode = async (page, filePath, props) => {
+  const fullPath = path.join(__dirname, "code", filePath);
+  try {
+    const code = fs.readFileSync(fullPath, "utf8");
+    const initialProps = props ? JSON.stringify(props) : "";
+
+    // Set code and initialProps attribute
+    await page.evaluate(
+      ({ code, initialProps }) => {
+        const viewer = document.querySelector("near-social-viewer");
```

Comment:
The original code was creating a new `new-social-viewer`element, while this new version is changing the existing. Do we know that changing attributes of the existing element triggers the same behaviour as creating a new?

Reply:
I had created the useCode function to support more complex cases, make code injection easier, and to address an issue that was arising while working on the Near.call tests: I noticed that modifying an existing element maintained the selectorPromise and allowed auth storage stages to work ~better~ (there are still some challenges-- and this happened during the "BOS is dead" talks, so had lost some motivation to support the regression tests -- I will write a comment there after this to detail these challenges).

Creating a new element via evaluate removes the selectorPromise -- how can we mock this?

I don't think this matters so much for the link tests or replacing evaluate in the vm/customelements test -- the tests pass the same before and after -- and there is still an existing test creating a new element.

I can write some tests for "code" attribute that verifies it works whether modifying an existing or creating new -- would that remedy the concern?

Reply:
yes that would fix it. probably not an issue at all, but then we are absolutely sure 👍

Reply:
Added tests for code and initial props

"should render code with props when creating a new element"
"should render code with props when changing attributes of an existing element"